src/templates/parseScriptTemplates.ts

```diff
@@ -162,16 +176,19 @@ export function parseScriptTemplate(rawTemplate: IRawTemplate, resources: IResou
     const userPromptedSettings: IFunctionSetting[] = [];
     if (rawTemplate.metadata.userPrompt) {
         for (const settingName of rawTemplate.metadata.userPrompt) {
-            const settings: IFunctionSetting[] | undefined = commonSettingsMap[functionConfig.inBindingType];
-            if (settings) {
-                const setting: IFunctionSetting | undefined = settings.find((bs: IFunctionSetting) => bs.name === settingName);
-                if (setting) {
-                    const functionSpecificDefaultValue: string | undefined = functionConfig.inBinding[setting.name];
-                    if (functionSpecificDefaultValue) {
-                        // overwrite common default value with the function-specific default value
-                        setting.defaultValue = functionSpecificDefaultValue;
+            if (functionConfig.inBinding) {
+                const inBinding: IFunctionBinding = functionConfig.inBinding;
+                const bindingTemplate: IBindingTemplate | undefined = bindingTemplates.find(b => b.type === inBinding.type);
+                if (bindingTemplate) {
+                    const setting: IFunctionSetting | undefined = bindingTemplate.settings.find((bs: IFunctionSetting) => bs.name === settingName);
+                    if (setting) {
+                        const functionSpecificDefaultValue: string | undefined = inBinding[setting.name];
+                        if (functionSpecificDefaultValue) {
+                            // overwrite common default value with the function-specific default value
+                            setting.defaultValue = functionSpecificDefaultValue;
+                        }
+                        userPromptedSettings.push(setting);
```

Comment:
Pushing `undefined` is intentional here, right?

Reply:
Uhhh no? I have `if (setting) {` a few lines up

Reply:
Ah yes, my bad. Misread the curlies ➰